Split Plan from caffe2.proto

houseroad opened the pull request:
C2 proto has no deps on Plan. Let's move it out.

Yangqing reviewed (approved):
There may be internal files that will need to have plan.pb.h added to their corresponding cc files, other than that it looks great to me.

houseroad committed:
Split Plan from caffe2.proto (#11306)

Summary:
C2 proto has no deps on Plan. Let's move it out.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/11306

Differential Revision: D9668785

fbshipit-source-id: f84ee383b3a567dd1161dbc4411cdc2d6f631adf

yf225 commented:
@houseroad Should we land this PR?

houseroad commented:
The plan changed. Close this diff at this moment.
